lib/flows/cron/data_retention_test.py

```diff
@@ -183,6 +184,149 @@ def testDeletesFlowsOlderThanGivenAge(self):
                         latest_timestamp - rdfvalue.Duration("150s"))
 
 
+class CleanTempTest(test_lib.FlowTestsBaseclass):
+  """Test the CleanTemp flow."""
+
+  NUM_TMP = 10
+
+  def setUp(self):
+    super(CleanTempTest, self).setUp()
+
+    self.tmp_urns = []
+    for i in range(self.NUM_TMP):
+      with test_lib.FakeTime(40 + 60 * i):
+        tmp_obj = aff4.FACTORY.Create("aff4:/tmp/%s" % i, "TempMemoryFile",
+                                      mode="rw", token=self.token)
+        self.tmp_urns.append(tmp_obj.urn)
+        tmp_obj.Close()
+
+  def testDoesNothingIfAgeLimitNotSetInConfig(self):
+    with test_lib.FakeTime(40 + 60 * self.NUM_TMP):
+      flow.GRRFlow.StartFlow(
+          flow_name=data_retention.CleanTemp.__name__,
+          sync=True, token=self.token)
+
+    tmp_urns = list(aff4.FACTORY.Open("aff4:/tmp",
+                                        token=self.token).ListChildren())
+    self.assertEqual(len(tmp_urns), 10)
+
+  def testDeletesTempWithAgeOlderThanGivenAge(self):
+    config_lib.CONFIG.Set("DataRetention.tmp_ttl",
```

Comment:
can you use a configoverrider context here? if you just set this, you will leak state. So if, for example, you run this test and afterwards the testDoesNothingIfAgeLimitNotSetInConfig above, the tmp_ttl will actually be set and the other test will fail. If you use a overrider context, it will restore the old settings after the test.